docs/pnp/migrate-vsto-to-office-web-add-in-shared-code-library-tutorial.md

```diff
@@ -0,0 +1,427 @@
+
+```csharp
+//Convert to Unicode listing
+    string result = "";
+    foreach (char c in cellValue)
+    {
+        int unicode = c;
+        result += $"{c}: {unicode}\r\n";
+    }
+```
+
+3. Update the line of code under the `//Output the result` comment to read as follows:
+
+```csharp
+//Output the result
+txtResult.Text = CellAnalyzerSharedLibrary.CellOperations.GetUnicodeFromText(cellValue);
+```
+
+6. From the **Debug** menu, choose **Start Debugging**. The custom task pane should still work as expected. Enter some text in a cell, and then test that you can convert it to a Unicode list with the add-in.
+
+## Create a REST API wrapper
+
+The VSTO Add-in can use the shared class library directly since they are both .NET projects. However the Office web add-in won't be able to use .NET since it uses JavaScript. Next you will need to create a REST API wrapper. This enables the Office web add-in to call a REST API, which then passes the call along to the shared class library.
+
+1. If you haven't already, start Visual Studio 2019 and open the **\start\Cell-Analyzer.sln** solution.
+2. Right-click the solution in **Solution Explorer** and choose **Add > New Project**.
+3. In the **Add a new project dialog**, choose **ASP.NET Core Web Application** and choose **Next**.
+4. In the **Configure your new project** dialog, set the following fields.
+    - Set the  **Project name** to **CellAnalyzerRESTAPI**.
+    - Leave the **Location** at it's default value.
+5. Choose **Create**.
+6. After the project is created, expand the **CellAnalyzerRESTAPI** project in **Solution Explorer**.
+7. Right-click **Dependencies** and choose **Add Reference**.
+8. Select **CellAnalyzerSharedLibrary** and choose **OK**.
+9. Right-click the **Controllers** folder, and choose **Add > Controller**.
+10. In the **Add New Scaffolded Item** dialog, choose **API Controller - Empty** and then **Add**.
+11. In the **Add Empty API Controller** dialog, name the controller **AnalyzeUnicodeController** and then choose **Add**.
+12. Open the **AnalyzeUnicodeController.cs** file and add the following code as a method to the `AnalyzeUnicodeController` class.
+
+```csharp
+[HttpGet]
+public ActionResult<string> AnalyzeUnicode(string value)
+{
+    if (value == null)
+    {
+        return BadRequest();
+    }
+    return CellAnalyzerSharedLibrary.CellOperations.GetUnicodeFromText(value);
+}
+```
+
+11. Right-click the CellAnalyzerRESTAPI project and choose **Set as Startup Project**.
+12. From the **Debug** menu, choose **Start Debugging**.
+13. A browser will launch. Enter the following URL to test that the REST API is working: **https://localhost:44323/api/analyzeunicode?value=test**. You should see a string returned with Unicode values for each character.
+
+## Create the Office web add-in
+
+When you create the Office web add-in, it will need to call the REST API. First you need to get the port number of the REST API server and save it.
```

Comment:
```suggestion
When you create the Office web add-in, it will make a call to the REST API. But first, you need to get the port number of the REST API server and save it for later.
```